sim/src/image.rs

```diff
@@ -135,6 +136,16 @@ impl ImagesBuilder {
             }
         }
 
+        if !unsupported_caps.contains(&Caps::SwapUsingScratch) {
+            let (_, scratch_len, _) = match areadesc.find(FlashId::ImageScratch) {
+                Some(info) => info,
+                None => return Err("insufficient partitions".to_string()),
+            };
+            if scratch_len < c::boot_trailer_sz(align as u32) as usize {
+                return Err("insufficient size for scratch partition".to_string());
+            }
```

Comment:
I've just attempted another run on `enc-aes256-rsa` without this filtering and the execution is halted by this assertion:
https://github.com/mcu-tools/mcuboot/blob/main/boot/bootutil/src/bootutil_misc.c#L147

The assertion ensures that the scratch area is larger than the trailer.
If I remove the assertion, the `enc-aes256-rsa` test still finishes successfully.
Should I remove the assertion?

Reply:
The assertion is correct for the trailer in the primary slot, probably have to update that function with a calculation of trailer size for scratch and use `fap` to determine which to use. You can leave as is, or fix it if you feel like it.

Reply:
I've tried the following patch to adjust the status offset to the size of the scratch area, but it didn't work and the `perm_with_fails` and `revert_with_fails` tests failed:
```patch
diff --git a/boot/bootutil/src/bootutil_misc.c b/boot/bootutil/src/bootutil_misc.c
index 67f0fe3..3bac13d 100644
--- a/boot/bootutil/src/bootutil_misc.c
+++ b/boot/bootutil/src/bootutil_misc.c
@@ -139,6 +139,11 @@ boot_status_off(const struct flash_area *fap)
 
     off_from_end = boot_trailer_sz(elem_sz);
 
+    if (flash_area_get_id(fap) == FLASH_AREA_IMAGE_SCRATCH) {
+        /* Image trailer may be larger than the scratch area */
+        off_from_end %= flash_area_get_size(fap);
+    }
+
     assert(off_from_end <= flash_area_get_size(fap));
     return flash_area_get_size(fap) - off_from_end;
 }
```

Reply:
I pushed two commits, one which removes that section of code above, that you can squash into the main commit it changes, and another with a fix to the boot_status_off function, which should correct the behavior and allow it to pass CI. Otherwise this looks pretty good and I will approve afterwards (assuming it passes CI!).